### PR TITLE
envoy: move listener manager code back under source

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: low
Testing: n/a
Docs Changes: n/a
Release Notes: n/a
Part of https://github.com/envoyproxy/envoy/issues/31113

### CODE REVIEW

**Review by ggreenway — COMMENTED** [review]
The new location `source/common/tcp` will now also contain the udp listener code; that's confusing. Maybe it should go into `source/common/listener_manager` instead?

/wait-any

File: test/per_file_coverage.sh [hunk 1]
```diff
@@ -59,6 +59,7 @@ declare -a KNOWN_LOW_COVERAGE=(
 "source/extensions/wasm_runtime/wasmtime:0.0" # Not enabled in coverage build
 "source/extensions/wasm_runtime/wavm:0.0" # Not enabled in coverage build
 "source/extensions/watchdog:83.3" # Death tests within extensions
+"source/extensions/listener_managers:70.5"
```
Comment (RyanTheOptimist) [inline test/per_file_coverage.sh:62]: Wow, this is quite low. It looks like the only change in that directory is a path in a #include. Do you understand why the test coverage is so low?

Comment (alyssawilk) [reply test/per_file_coverage.sh:62]: this directory covered the highly tested TCP code and the not highly tested validation manager code.

**Review by ggreenway — CHANGES_REQUESTED** [review]
I think the files previously in source/common/tcp were accidentally duplicated into listener_manager: async_tcp_client_impl.h/cc and conn_pool.h/cc

/wait

**Review by ggreenway — APPROVED** [review]

**Pull request merged**